nav2_smac_planner/src/a_star.cpp

```diff
@@ -294,6 +295,11 @@ bool AStarAlgorithm<NodeT>::createPath(
       }
     }
 
+    // Check for cancel every Nth iteration
+    if (iterations % _cancel_check_interval == 0 && cancel_checker()) {
```

Comment:
Is 500 vs 5000 a big enough difference to not have it in the same timing interval branch (and possibly rename to be more generic)?

Reply:
I am not very sure about numbers so I am open to suggestions for default values, but I made it a parameter now so seperate branch is needed.

Reply:
Lets not then, use the same parameter branch to check for all potential preemption conditions like cancel / timeout